File: INFS4102_project-main/Backend/eshop/src/main/java/qa/udst/eshop/models/EquipmentProduct.java
package qa.udst.eshop.models;

import jakarta.validation.constraints.*;


//Equipment Model: + Weight and Dimensions
public class EquipmentProduct extends Product{
    @Min(0)
    @NotNull
    private double weightKg;

    @NotBlank
    private String dimensions;
    
    public EquipmentProduct(String name, double price, ProductCategory category, String desc,
            @Min(0) @NotNull double weightKg, @NotBlank String dimensions, @NotBlank String imageURL) {
        super(name, price, category, desc, imageURL);
        this.weightKg = weightKg;
        this.dimensions = dimensions;
    }

    public double getWeightKg() {
        return weightKg;
    }

    public void setWeightKg(double weightKg) {
        this.weightKg = weightKg;
    }

    public String getDimensions() {
        return dimensions;
    }

    public void setDimensions(String dimensions) {
        this.dimensions = dimensions;
    }

    
    
    

    


}

File: INFS4102_project-main/Frontend/eshop_ui/lib/models/EquipmentProduct.dart
import 'Product.dart';

class EquipmentProduct extends Product {
  final double weightKg;
  final String dimensions;

  const EquipmentProduct({
    required super.id,
    required super.name,
    required super.price,
    required super.category,
    required super.desc,
    required super.imageURL,
    required this.weightKg,
    required this.dimensions,
  });

  //converts json
  factory EquipmentProduct.fromJson(Map<String, dynamic> json) {
    return EquipmentProduct(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] as num).toDouble(),
      category: json['category'] ?? '',
      desc: json['desc'] ?? '',
      imageURL: json['imageURL'] ?? '',
      weightKg: (json['weightKg'] ?? 0).toDouble(),
      dimensions: json['dimensions'] ?? '',
    );
  }
}
